src/Features/Core/Portable/ChangeSignature/AbstractChangeSignatureService.cs

```diff
@@ -807,16 +808,64 @@ protected static int GetParameterIndex<TNode>(SeparatedSyntaxList<TNode> paramet
                             continue;
                         }
 
+                        SyntaxNode expression = null;
+
+                        var callsiteInferenceFailed = false;
+
+                        if (addedParameter.CallSiteKind == CallSiteKind.Inferred && addedParameter.TypeBinds && getSemanticModelAndRecommendations != null)
+                        {
+                            callsiteInferenceFailed = true;
+
+                            (var semanticModel, var recommendations) = await getSemanticModelAndRecommendations().ConfigureAwait(false);
+
+                            var targetType = addedParameter.Type;
+
+                            var sourceSymbols = recommendations.Where(r => r.IsNonImplicitAndFromSource());
+                            var sourceSymbolsWithTypes = sourceSymbols.Where(s => s.GetSymbolType() != null);
+                            var orderedSourceSymbolsWithTypes = sourceSymbolsWithTypes.OrderByDescending(s => s.Locations.First().SourceSpan.Start);
```

Comment:
this is a weird way to order since you're comparing by location, but the results might be in differerent files.  this doesn' teally make sense IMO.

Reply:
I've tried to make this clearer. 

> results might be in differerent files

Are you talking about partial classes? Because otherwise it's limited to fields/properties/locals/parameters, so we should be fine there I think?